tsl/src/compression/create.c

```diff
@@ -996,3 +1051,29 @@ tsl_process_compress_table(AlterTableCmd *cmd, Hypertable *ht,
 	/* do not release any locks, will get released by xact end */
 	return true;
 }
+
+/* Add a column to a table that has compression enabled
+ * This function specifically adds the column to the internal compression table.
+ */
+void
+tsl_process_compress_table_add_column(Hypertable *ht, ColumnDef *orig_def)
+{
+	struct CompressColInfo compress_cols;
+	Oid coloid;
+	int32 orig_htid = ht->fd.id;
+	char *colname = orig_def->colname;
+	TypeName *orig_typname = orig_def->typeName;
+
+	coloid = LookupTypeNameOid(NULL, orig_typname, false);
+	compresscolinfo_init_singlecolumn(&compress_cols, colname, coloid);
+	if (TS_HYPERTABLE_HAS_COMPRESSION_TABLE(ht))
+	{
+		int32 compress_htid = ht->fd.compressed_hypertable_id;
+		Hypertable *compress_ht = ts_hypertable_get_by_id(compress_htid);
+		add_column_to_compression_table(compress_ht, &compress_cols);
+	}
+	else
+		Assert(TS_HYPERTABLE_HAS_COMPRESSION_ENABLED(ht));
```

Comment:
I believe it might fail if the extension will be build with `-DNDEBUG`  flag (or maybe not, if `;` will be kept), anyway I suggest to wrap this statement into the code block

Reply:
added